fxa/oauth.py

```diff
@@ -281,6 +281,11 @@ def verify_token(self, token, scope=None):
                         # (e.g. it's expired) then using a different key won't
                         # help.
                         continue
+                else:
+                    # It's a well-formed JWT, but not signed by any of the advertized keys.
+                    # We can immediately surface this as an error.
+                    if len(keys) > 0:
+                        raise TrustError({"error": "invalid signature"})
```

Comment:
Changing the tests to expect a `TrustError` rather than a generic `Exception` helped me identify another opportunity to avoid falling back to the `/verify` endpoint.

Reply:
I think you already wanted this error before but it didn't make it into the current wacky control flow graph.